### DISCUSSION TITLE
Sonic Mega Collection Plus (PS2) files?

### DISCUSSION THREAD

**Discussion by TheNametag** [farmerbb/RED-Project#51] [Ideas]
This is a much older set, and may be well outside the purview of this project, but I was wondering if it would be possible to extract the ROMs from the data files of the Sonic Mega Collection Plus PS2 discs.  Specifically the PS2 disc is readable on PCs, so the files are visible, but encrypted.  I've seen methods to extract the ROMs using emulators and extracting the data from the emulator's memory, but that doesn't give access to all of the ROMs on the disc.  The files contain ROMs from multiple regions, whereas running the game only nets you one region.  (I do believe there is a mod for the PC version which lets you switch regions, but I don't have that so I cannot confirm.)

The files have the `.CPT` file extension..  This is a type of encrypted compression that, it appears, may be unlockable with a password if the password is known.  It may be possible to find the key in emulation memory when it is decrypting the files.  `.CPT `is an archive format used by `ccrypt`, which may be what is used here.

The disc contains 7 files in the `GGROM` folder:
```
G_SONIC.CPT
LABYLIN.CPT
MBMACHIN.CPT
SONIC.CPT
SONIC_CH.CPT
SONICDRI.CPT
SONICTAI.CPT
```

It contains 17 files in each of the following folders: `MDROM`, `MDROM_EU` `MDROM_US`

```
COMIX.CPT
FLICKY.CPT
KNU_P1.CPT
KNU_P2.CPT
KNU_P3.CPT
OOZE.CPT
PUYO_USA.CPT
RISTER.CPT
S_SPIN.CPT
SANDK.CPT
SONIC1.CPT
SONIC1B.CPT
SONIC1U.CPT
SONIC2.CPT
SONIC2B.CPT
SONIC3.CPT
SONIC3D.CPT
```

The files in each folder share a name, but have different MD5 hashes.  I am not sure if they are different ROMs for different regions or if they have different decryption keys.

I realize this is likely low priority, given that the Game Gear ROMs are widely accessible in Sonic Adventure DX on Steam, and the Mega Drive/Genesis games are also, mostly, available on Steam (though they have removed the Sonic games recently, and this set was unique in that it had the linked and unlinked Sonic and Knuckles games).  Plus my clues may be red herrings, with ccrypt being completely incorrect for these files.  But I thought I would put it out there with the specific files readable on the disc.

**Comment by PW5190** [comment]
What about the Xbox, PC, and the GameCube (non-Plus) versions?

**Comment by TheNametag** [comment]
GamecCube version shouldn't have the files, that was the whole point of Plus.  GameCube those files were on Sonic Adventure DX instead.  I cannot confirm anything for other versions.

**Reply by PW5190** [reply]
I know, the GameCube version had the ROMs in .dat files.

**Reply by TheNametag** [reply]
Right, right, I was only thinking of the GameGear games.  The other ROMs should be in the files somewhere.  I don't have a way to read a GameCube disc to PC though.

**Reply by PW5190** [reply]
> Right, right, I was only thinking of the GameGear games. The other ROMs should be in the files somewhere. I don't have a way to read a GameCube disc to PC though.

You do not actually need to put a GameCube disc into your PC for this. All you need is a softmodded Wii or a GameCube capable of launching homebrew, and a tool called CleanRip or USB Loader GX (Wii only) to dump the disc to either SD or USB (Or SD Gecko or SD2SP2 if using CleanRip on a GameCube). You can then load up the ISO in Dolphin, and check the filesystem.

**Comment by TheNametag** [comment]
Something to note with those ROMs, they include 3 versions of Sonic 1, and 2 versions of Sonic 2.  Now, I know there's codes for the original Gamecube version to unlock two other versions of Sonic 1, but I've only seen codes for 1 other version of Sonic 1 in the PS2 collection, and neither collection has known codes to unlock a second Sonic 2 version.

**Comment by kjetil-f** [comment]
About the Sonic Adventure DX on Steam.  I don't think the ROMs is included originally. I know about the BetterSADX patch, but doesn't it just add pirated versions of the Game Gear games back to the game?

**Comment by TheNametag** [comment]
They are on the Steam version.  You can read extraction instructions [here.](https://legal-roms.fandom.com/wiki/List_of_Game_Gear_ROMs#Sonic_Adventure_DX)

**Reply by kjetil-f** [reply]
Thank you. Guess my assumption was wrong. Tried it myself, and it worked.

**Comment by DrRetroMod** [comment]
So I am just seeing this now. I spent a lot of time over the last few weeks extracting different games. I have succesfully extracted all the games from the PS2 release of Sonic Mega Collection Plus. They way to do it is by using PCSX2 and save states. When you use save states it creates a memory dump of what is loaded at that time. I used this to dump the roms. I would start the game, load a rom on the game (wait till game is playable), and the save state it. I was then able to hex edit out all the roms for the memory by using HxD. Tok a bit of time getting used to it, but I managed in the end. Most are 100% recognisable dumps, except 'sonic & knuckle + sonic hedgehog 2' and also 'sonic & knuckle + sonic hedgehog 3'. I think this is in part because the two roms that are used are different region match to others previously made.

I am planning to do a full write up on this soon, so if you are still interested let me know 🙏

**Reply by SlashEraser** [reply]
> I am planning to do a full write up on this soon, so if you are still interested let me know 🙏

Oh nice.
